apps/condo/domains/news/components/NewsForm/InputStep.tsx

```diff
@@ -0,0 +1,810 @@
+import { NewsItemScopeNoInstanceType } from '../types'
+
+const buildCounterStyle = (textLength: number, type: 'Body' | 'Title'): React.CSSProperties => {
+    const style: React.CSSProperties = {
+        position: 'absolute',
+        right: 0,
+        margin: '12px',
+        padding: '2px 10px',
+        borderRadius: '100px',
+        backgroundColor: `${colors.gray[7]}`,
+    }
+
+    if (textLength > 0) {
+        style.backgroundColor = `${colors.black}`
+    }
+
+    if (type === 'Body') {
+        style.bottom = '12px'
+    }
+    if (type === 'Title') {
+        style.bottom = '12px'
+    }
+
+    return style
+}
+
+export type SharingAppValues = {
+    formValues: Record<string, unknown>
+    preview: {
+        renderedTitle: string
+        renderedBody: string
+    }
+    isValid: boolean
+}
+
+interface INewsItemSharingForm {
+    sharingAppData?: {
+        app: B2BApp
+        id: string
+        newsSharingConfig: B2BAppNewsSharingConfig
+    }
+    onSubmit: (SharingAppValues) => void
+    onSkip: (SharingAppValues) => void
+
+    initialValues: SharingAppValues | undefined
+
+    newsItemData: {
+        type: string
+        validBefore?: string
+        title: string
+        body: string
+    }
+}
+
+export type BaseNewsFormProps = {
+    organizationId: string
+    initialValues?: Partial<INewsItem>
+    & Partial<{
+        newsItemScopes: INewsItemScope[]
+        hasAllProperties: boolean
+        sendPeriod: SendPeriodType
+        properties?: IProperty[]
+    }>
+    templates: { [key: string]: {
+        title: string
+        body: string
+        type: string | null
+        id?: string
+        label?: string
+        category?: string
+    } }
+    totalProperties: number
+    autoFocusBody?: boolean
+}
+
+type TInputStepProps = INewsItemSharingForm & BaseNewsFormProps & {
+    form: any
+    scope: TScope
+    setScope:  React.Dispatch<React.SetStateAction<TScope>>
+    isSharing: boolean
+    selectedProperty: {
+        loading: boolean
+        objs:  IProperty[]
+    }
+    initialPropertyIds: string[]
+    initialFormValues: Record<string, unknown>
+}
+
+export const FormContainerContainer: React.FC = ({ children }) => {
+    return (
+        <Col span={24} style={BIG_MARGIN_BOTTOM_STYLE}>
+            <Row gutter={BIG_HORIZONTAL_GUTTER}>
+                {children}
+            </Row>
+        </Col>
+    )
+}
+
+export const InputStep: React.FC<TInputStepProps> = ({
+    form,
+    scope,
+    onSkip,
+    onSubmit,
+    setScope,
+    templates,
+    isSharing,
+    newsItemData,
+    initialValues,
+    autoFocusBody,
+    sharingAppData,
+    organizationId,
+    totalProperties,
+    selectedProperty,
+    initialPropertyIds,
+    initialFormValues,
+}
+) => {
+    const intl = useIntl()
+    const { app: sharingApp, id: ctxId } = sharingAppData ?? { ctxId: null, sharingApp: null }
+    const { id, newsSharingConfig } = sharingApp ?? { id: null, newsSharingConfig: null }
```

Comment:
I do not really like that we have one big component that renders both sharing app and not sharing app. Does not look clean to me

I'd like to propose something like this:

BaseNewsForm - handles steps and skips logic
- InputStep – just takes building blocks and compiles them in single component 
- - bodyControl/IFrameBodyControl.tsx
- - bodyControl/CondoBodyControl.tsx
- - scopeControl/CondoScopeControl.tsx
- - scopeControl/GetRecipientsScopeControl.tsx
- - RecipientsDisplay.tsx
- - PreviewDisplay.tsx

We can discuss this